zaza/openstack/charm_tests/cinder_backup/tests.py

```diff
@@ -0,0 +1,182 @@
+        pools = ceph_utils.get_ceph_pools(unit_name)
+        expected_pool = 'cinder-ceph'
+        cinder_ceph_pool = pools[expected_pool]
+
+        # Check ceph cinder pool object count, disk space usage and pool name
+        logging.info('Checking ceph cinder pool original samples...')
+        pool_name, obj_count, kb_used = ceph_utils.get_ceph_pool_sample(
+            unit_name, cinder_ceph_pool)
+
+        obj_count_samples.append(obj_count)
+        pool_size_samples.append(kb_used)
+
+        self.assertEqual(pool_name, expected_pool)
+
+        # Create ceph-backed cinder volume
+        cinder_vol = openstack_utils.create_volume(
+            self.cinder_client,
+            name='{}-410-vol'.format(self.RESOURCE_PREFIX),
+            size=1)
+        # Backup the volume
+        vol_backup = openstack_utils.create_volume_backup(
+            self.cinder_client,
+            cinder_vol.id,
+            name='{}-410-backup-vol'.format(self.RESOURCE_PREFIX))
+        # Delete the volume
+        openstack_utils.delete_volume(self.cinder_client, cinder_vol.id)
+        # Restore the volume
+        self.cinder_client.restores.restore(vol_backup.id)
+        openstack_utils.resource_reaches_status(
+            self.cinder_client.backups,
+            vol_backup.id,
+            expected_status='available',
+            msg='Backup status wait')
+        # Delete the backup
+        openstack_utils.delete_volume_backup(self.cinder_client, vol_backup.id)
+
+        # Re-check ceph cinder pool object count and disk usage
+        time.sleep(10)
+        logging.info('Checking ceph cinder pool samples '
+                     'after volume create...')
+        pool_name, obj_count, kb_used = ceph_utils.get_ceph_pool_sample(
+            unit_name, cinder_ceph_pool, self.model_name)
+
+        obj_count_samples.append(obj_count)
+        pool_size_samples.append(kb_used)
+
+        name = '{}-410-vol'.format(self.RESOURCE_PREFIX)
+        vols = self.cinder_client.volumes.list()
+        try:
+            cinder_vols = [v for v in vols if v.name == name]
+        except AttributeError:
+            cinder_vols = [v for v in vols if v.display_name == name]
+        if not cinder_vols:
+            # NOTE(hopem): it appears that at some point cinder-backup stopped
+            # restoring volume metadata properly so revert to default name if
+            # original is not found
+            name = "restore_backup_{}".format(vol_backup.id)
+            try:
+                cinder_vols = [v for v in vols if v.name == name]
+            except AttributeError:
+                cinder_vols = [v for v in vols if v.display_name == name]
+
+        self.assertTrue(cinder_vols)
+
+        cinder_vol = cinder_vols[0]
+
+        # Delete restored cinder volume
+        openstack_utils.delete_volume(self.cinder_client, cinder_vol.id)
+
+        # Final check, ceph cinder pool object count and disk usage
+        time.sleep(10)
```

Comment:
Same comment as before, look into replacing the sleep with a call to resource_removed